src/main/java/org/sagebionetworks/web/client/widget/display/ProjectDisplayDialog.java

```diff
@@ -0,0 +1,139 @@
+	public static final String DOCKER = "Docker";
+	
+	@Inject
+	public ProjectDisplayDialog(ProjectDisplayView view,
+			SynapseClientAsync synapseClient, 
+			SynapseAlert synAlert,
+			ClientCache storage, 
+			GlobalApplicationState globalAppState) {
+		this.view = view;
+		this.synapseClient = synapseClient;
+		this.synAlert = synAlert;
+		this.storage = storage;
+		this.globalAppState = globalAppState;
+		view.setSynAlertWidget(synAlert);
+		view.setPresenter(this);
+	}
+
+	public void configure(String projectId, String userId) {
+		this.projectId = projectId;
+		this.userId = userId;
+		this.tag = this.userId + "_" + this.projectId + "_";
+	}
+	
+	public Widget asWidget() {
+		return view.asWidget();
+	}
+	
+	public void show() {
+		synapseClient.getProjectDisplay(projectId, new AsyncCallback<ProjectDisplayBundle>() {
+			@Override
+			public void onFailure(Throwable caught) {
+				view.showErrorMessage(caught.getMessage());
+			}
+
+			@Override
+			public void onSuccess(ProjectDisplayBundle result) {
+				ProjectDisplayDialog.this.result = result;
+				boolean wiki = Boolean.parseBoolean(storage.get(tag + WIKI)) || result.wikiHasContent();
+				view.setWiki(wiki);
+				boolean files = Boolean.parseBoolean(storage.get(tag + FILES)) || result.filesHasContent();
+				view.setFiles(files);
+				boolean tables = Boolean.parseBoolean(storage.get(tag + TABLES)) || result.tablesHasContent();
+				view.setTables(tables);
+				boolean challenge = Boolean.parseBoolean(storage.get(tag + CHALLENGE)) || result.challengeHasContent();
+				view.setChallenge(challenge);
+				boolean discussion = Boolean.parseBoolean(storage.get(tag + DISCUSSION)) || result.discussionHasContent();
+				view.setDiscussion(discussion);
+				boolean docker = Boolean.parseBoolean(storage.get(tag + DOCKER)) || result.dockerHasContent();
+				view.setDocker(docker);
+				view.show();
+			}
+			
+		});
+		
+	}
+	
+	public void hide() {
+		view.hide();
+		globalAppState.getPlaceChanger().goTo(new Synapse(projectId));
+	}
+	
+	public void clear() {
+		synAlert.clear();
+		view.clear();
+	}
+
+	@Override
+	public void onSave() {
+		synAlert.clear();
+		
+		//validate that at least one is selected
+		if (! (view.getWiki() || view.getFiles() || view.getTables() || view.getChallenge() || view.getDiscussion() || view.getDocker())) {
+			synAlert.showError("Please select at least one feature.");
+			return;
+		}
+		
+		try {
+			updateCache(view.getWiki(), result.wikiHasContent(), WIKI);
+			updateCache(view.getFiles(), result.filesHasContent(), FILES);
+			updateCache(view.getTables(), result.tablesHasContent(), TABLES);
+			updateCache(view.getChallenge(), result.challengeHasContent(), CHALLENGE);
+			updateCache(view.getDiscussion(), result.discussionHasContent(), DISCUSSION);
+			updateCache(view.getDocker(), result.dockerHasContent(), DOCKER);
+		} catch (IllegalArgumentException e) {
+			synAlert.showError(e.getMessage());
+			return;
+		}
+		hide();
+	}
+	
```

Comment:
They will be presented with the project display settings again (on the other machine)